api/src/main/java/com/epam/pipeline/manager/preprocessing/SampleSheetParser.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright 2022 EPAM Systems, Inc. (https://www.epam.com/)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.epam.pipeline.manager.preprocessing;
+
+import com.epam.pipeline.entity.samplesheet.SampleSheet;
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+import org.springframework.data.util.Pair;
+import org.springframework.util.Assert;
+import org.springframework.util.StringUtils;
+
+import java.io.BufferedReader;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * This class parses provided content regarding to sample sheet description:
+ * https://www.illumina.com/content/dam/illumina-marketing/documents/products/technotes/sequencing-sheet-format-specifications-technical-note-970-2017-004.pdf
+ * */
+@SuppressWarnings({"LineLength", "HideUtilityClassConstructor"})
```

Comment:
Why do we suppress `LineLength` warning?

Reply:
because of long javaDoc comment line with link included